Rearrange device events in ReceiveIGTLMessage and add AboutToReceiveEvent

Sunderlandkyl opened the pull request:
- ReceiveIGTLMessage() now has only one exit point where it returns success or failure.
- Modified and DeviceXYZModifiedEvents are triggered if fromIGTL() is successful.
- ReceiveEvent is invoked whether or not there are issues with fromIGTL() conversion.
- AboutToReceiveEvent is useful for applications such as Slicer that would like to be notified before device content is modified.

lassoan commented on Devices/igtlioDevice.h:
```diff
@@ -52,6 +52,7 @@ class OPENIGTLINKIO_DEVICES_EXPORT igtlioDevice : public vtkObject
  enum {
    ReceiveEvent          = 118948,
    ResponseEvent         = 118952,
+   AboutToReceiveEvent,
```
maybe we could put it before ReceiveEvent with an ID of 118947 (to match the order how they are invoked)

lassoan commented on Devices/igtlioImageDevice.cxx:
```diff
@@ -76,15 +76,16 @@ igtlioImageConverter::ContentData igtlioImageDevice::GetContent()
-   }
-
- return 0;
+  this->InvokeEvent(AboutToReceiveEvent, this);
```
Could we move invoking AboutToReceiveEvent and ReceiveEvent to the caller of ReceiveIGTLMessage (in igtlioConnector)?

Sunderlandkyl commented:
AboutToReceiveEvent and ReceiveEvent are now invoked in igtlioConnector.
Event is still invoked on the igtlioDevice.

Sunderlandkyl committed:
Rearrange device events in ReceiveIGTLMessage() and add AboutToReceiveEvent

- ReceiveIGTLMessage() now has only one exit point where it returns success or failure.
- Modified and DeviceXYZModifiedEvents are triggered if fromIGTL() is successful.
- ReceiveEvent is invoked whether or not there are issues with fromIGTL() conversion.
- AboutToReceiveEvent is useful for applications such as Slicer that would like to be notified before device content is modified.
